Realm/RLMRealm.h

```diff
@@ -225,20 +225,41 @@ NS_ASSUME_NONNULL_BEGIN
 
  The destination file cannot already exist.
 
- Note that if this method is called from within a write transaction, the
- *current* data is written, not the data from the point when the previous write
+ Note that if this method is called from within a write transaction,
+ the *current* data is written, not the data from the point when the previous write
  transaction was committed.
 
  @param fileURL Local URL to save the Realm to.
  @param key     Optional 64-byte encryption key to encrypt the new file with.
- @param error   If an error occurs, upon return contains an `NSError` object
- that describes the problem. If you are not interested in
- possible errors, pass in `NULL`.
+ @param error  If an error occurs, upon return contains an `NSError` object
+              that describes the problem. If you are not interested in
+              possible errors, pass in `NULL`.
 
  @return `YES` if the Realm was successfully written to disk, `NO` if an error occurred.
  */
 - (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(nullable NSData *)key error:(NSError **)error;
 
+/**
+ Writes a compacted and optionally encrypted copy of thre Realm to the given local URL.
+
+ The destination file cannot already exist.
+
+ Note that if this method is called from within a write transaction,
+ the *current* data is written, not the data from the point when the previous write
+ transaction was committed.
+
+ @param fileURL Local URL to save the Realm to.
+ @param key     Optional 64-byte encryption key to encrypt the new file with.
+ @param enableHistory  Indicates whether the copied Realm will have sync history enabled
+                      and synchronize data with the Realm server. All changes from the
+                      original realm must be uploaded to the server prior to calling
+                      `writeCopyToURL`, otherwise and error is thrown. If `enableHistory`
+                      is false, the copied Realm functions as a local realm.
+ @param error  If an error occurs, upon return contains an `NSError` object
+              that describes the problem. If you are not interested in
+              possible errors, pass in `NULL`.
+ */
+- (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(nullable NSData *)key enableHistory:(BOOL)enableHistory error:(NSError **)error;
```

Comment:
"History" as a concept is sort of an implementation detail. The thing the user cares about is that the end result can be opened as a sync Realm, not the specific details about what's different about sync/non-sync Realms.

I don't have great ideas for what the argument name should be for that though. The obvious things like `enableSync:` or `synchronizedRealm:` seem bad.

Reply:
I guess `writeSynchronizableCopyToURL` is a mouthful, but also reasonably descriptive.

Reply:
In the latest commit I did go with `enableSync` param. I think a new method `writeSyncCopyToURL` might be alright.
What do you think?